test/priceOracle.test.ts

```diff
@@ -0,0 +1,124 @@
+            await deploy();
+            expect(_oracle.address).to.be.properAddress;
+        });
+        it("Should init correct data on deployment", async function () {
+            let beraBytes = toUtf8Encode("🐻");
+            let licenseBytes = toUtf8Encode("🪪");
+            expect(await _oracle.isEmoji(beraBytes)).to.eq(true);
+            expect(await _oracle.isEmoji(licenseBytes)).to.eq(true);
+        });
+    });
+
+    describe("Updates", function () {
+        describe("setEmojis", function () {
+            it("Should set emojis", async function () {
+                let emojis = ["🐻‍❄️", "🫶"];
+                await _oracle.setEmojis(emojis);
+
+                expect(await _oracle.isEmoji(toUtf8Encode("🐻‍❄️"))).to.eq(true);
+                expect(await _oracle.isEmoji(toUtf8Encode("🫶"))).to.eq(true);
+            });
+
+            it("Should revert if caller is not an owner", async function () {
+                await expect(_oracle.connect(_otherAccount).setEmojis(["🦶"])).to.be.revertedWith(
+                    "Ownable: caller is not the owner"
+                );
+            });
+        });
+
+        describe("countEmojis", function () {
+            it("Should successfully count emojis", async function () {
+                expect(await _oracle.countEmojis(["🐻", "🪪"])).to.eq(2);
+            });
+            it("Should revert if incorrect input", async function () {
+                await expect(_oracle.countEmojis(["🐻‍❄️", "🫶"])).to.be.revertedWithCustomError(_oracle, "Nope");
+            });
+        });
+
+        describe("dollarPriceForNamePerYear", function () {
+            it("Should return a correct price", async function () {
+                let emojis = ["🐻‍❄️", "🫶", "🎃", "💩", "😮"];
+                await _oracle.setEmojis(emojis);
+                expect(await _oracle.dollarPriceForNamePerYear(["🐻‍❄️", "🫶"])).to.eq(BigInt((690 * 1e18 * 69) / 100));
+                expect(await _oracle.dollarPriceForNamePerYear(["🐻‍❄️", "🫶", "🎃"])).to.eq(
+                    BigInt((450 * 1e18 * 69) / 100)
+                );
+                expect(await _oracle.dollarPriceForNamePerYear(["🫶", "🐻‍❄️", "🎃", "💩"])).to.eq(
+                    BigInt((80 * 1e18 * 69) / 100)
+                );
+                expect(await _oracle.dollarPriceForNamePerYear(["🐻‍❄️", "🫶", "🎃", "💩", "😮"])).to.eq(
+                    BigInt((25 * 1e18 * 69) / 100)
+                );
+                expect(await _oracle.dollarPriceForNamePerYear(["b", "e", "r", "a", "🐻‍❄️"])).to.eq(
+                    BigInt((25 * 1e18 * 25) / 100)
+                );
+                expect(await _oracle.dollarPriceForNamePerYear(["o"])).to.eq(BigInt(999 * 1e18));
+            });
+
+            it("Should revert if input length == 1", async function () {
+                await expect(_oracle.dollarPriceForNamePerYear(["🐻"])).to.be.revertedWithCustomError(_oracle, "Nope");
+            });
+        });
+
+        describe("price", function () {
```

Comment:
I reckon we should have a test for name lengths from 1 to 5 with and without emojis